### PR TITLE
test(manager/helmv3): migrate from fs.readFile to fs.readLocalFile 

### PR DESCRIPTION
**Pull request by olegkrivtsov**
## Changes:

migrate from fs.readFile to fs.readLocalFile (helmv3) #12987 

## Context:

Part of #7049, related to PR https://github.com/renovatebot/renovate/pull/12987

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/helmv3/__snapshots__/artifacts.spec.ts.snap [hunk 1]
```diff
@@ -43,7 +43,7 @@ Array [
     },
   },
   Object {
-    "cmd": "helm dependency update --registry-config /tmp/renovate/cache/__renovate-private-cache/registry.json --repository-config /tmp/renovate/cache/__renovate-private-cache/repositories.yaml --repository-cache /tmp/renovate/cache/__renovate-private-cache/repositories ''",
+    "cmd": "helm dependency update --registry-config registry.json --repository-config repositories.yaml --repository-cache repositories undefined",
```
Comment (viceice) [inline lib/manager/helmv3/__snapshots__/artifacts.spec.ts.snap:46]: Undefined is bad here

File: lib/manager/helmv3/artifacts.spec.ts [hunk 2]
```diff
@@ -1,24 +1,17 @@
...
 
-jest.mock('fs-extra');
 jest.mock('child_process');
 jest.mock('../../util/exec/env');
 jest.mock('../../util/http');
-
-const fs: jest.Mocked<typeof _fs> = _fs as any;
-const exec: jest.Mock<typeof _exec> = _exec as any;
-const env = mocked(_env);
+jest.mock('../../util/fs');
```
Comment (viceice) [inline lib/manager/helmv3/artifacts.spec.ts:14]: I think we should use our new Fixtures class to mock files, can you check other managers for samples?

**Review by viceice — COMMENTED** [review]
Needs lint fixes too

**Comment by olegkrivtsov** [comment]
Hi @viceice the build has failed on `e2e` tests, but I'm not sure how to fix the problem. Can you advice?

**Review by viceice — COMMENTED** [review]
I think it was a temporary issue. This needs deconflicting now.

**Review by viceice — COMMENTED** [review]
Needs snapshot updates

File: lib/manager/helmv3/artifacts.spec.ts [hunk 3]
```diff
@@ -64,10 +60,14 @@ describe('manager/helmv3/artifacts', () => {
...
   });
+
   it('returns null if unchanged', async () => {
-    fs.readFile.mockResolvedValueOnce(ociLockFile1 as any);
+    fs.readLocalFile.mockResolvedValueOnce(ociLockFile1 as any);
+    fs.getSiblingFileName.mockReturnValueOnce('Chart.lock');
     const execSnapshots = mockExecAll(exec);
-    fs.readFile.mockResolvedValueOnce(ociLockFile1 as any);
+    fs.readLocalFile.mockResolvedValueOnce(ociLockFile1 as any);
+    fs.privateCacheDir.mockReturnValue('');
```
Comment (viceice) [inline lib/manager/helmv3/artifacts.spec.ts:69]:
```suggestion
    fs.privateCacheDir.mockReturnValue('/tmp/renovate/cache/__renovate-private-cache');
```
This should reduce snapshot changes.
Can you add this to the other mocks too?

**Comment by olegkrivtsov** [comment]
Hi @viceice I see some strange CodeQL build error, but I have no idea what could create it...

**Review by viceice — APPROVED** [review]

**Pull request merged**